src/server/initServer.ts

```diff
@@ -267,13 +263,144 @@ const initServer = (
           );
         }
 
+        // Build MongoDB query based on filters
+        const query: any = {};
+
+        /* -------------- Date Filter ------------- */
+        const { startDate, endDate } = filters.dateFilterState;
```

Comment:
Let's cast filters to the correct type (use same type as client, add to server and import it here).